Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java

```diff
@@ -740,10 +761,70 @@ public boolean hasSearchQuery() {
         return mSearchString != null && !mSearchString.equals("");
     }
 
+    public void addSearchItem(String item, int index) {
+        Preferences preferences = getPreferences();
+
+        if (preferences != null) {
+            List<String> recents = preferences.getRecentSearches();
+            recents.remove(item);
+            recents.add(index, item);
+            // Trim recent searches to MAX_RECENT_SEARCHES (currently 5) if size is greater than MAX_RECENT_SEARCHES.
+            preferences.setRecentSearches(recents.subList(0, recents.size() > MAX_RECENT_SEARCHES ? MAX_RECENT_SEARCHES : recents.size()));
```

Comment:
it was a separate thing and may have been lost, but we can collapse this ternary expression into a single declaration…

```diff
- preferences.setRecentSearches(recents.subList(0, recents.size() > MAX_RECENT_SEARCHES ? MAX_RECENT_SEARCHES : recents.size()));
+ preferences.setRecentSearches(recents.subList(0, Math.min(MAX_RECENT_SEARCHES, recents.size())));
```

Reply:
Interestingly, a ternary expression is used in the `Math.min(int a, int b)` source code.

```java
    /**
     * Returns the smaller of two {@code int} values. That is,
     * the result the argument closer to the value of
     * {@link Integer#MIN_VALUE}.  If the arguments have the same
     * value, the result is that same value.
     *
     * @param   a   an argument.
     * @param   b   another argument.
     * @return  the smaller of {@code a} and {@code b}.
     */
    public static int min(int a, int b) {
        return (a <= b) ? a : b;
    }
```

Since they are the nearly the exact same thing, there doesn't seem like much advantage other than developer preference.  Using the ternary expression could help reviewers or future maintainers understand the logic a little easier.  That's also why I added the comment above that statement to explain what's going on.